engine/src/main/java/io/camunda/zeebe/engine/processing/usertask/UserTaskCommandPreconditionChecker.java

```diff
@@ -52,13 +54,25 @@ protected Either<Tuple<RejectionType, String>, UserTaskRecord> check(
 
     final LifecycleState lifecycleState = userTaskState.getLifecycleState(userTaskKey);
 
-    if (validLifecycleStates.contains(lifecycleState)) {
-      return Either.right(persistedRecord);
+    if (!validLifecycleStates.contains(lifecycleState)) {
+      return Either.left(
+          Tuple.of(
+              RejectionType.INVALID_STATE,
+              String.format(INVALID_USER_TASK_STATE_MESSAGE, intent, userTaskKey, lifecycleState)));
+    }
+
+    if (intent.equals("claim")) {
+      final String assignee = persistedRecord.getAssignee();
+      final boolean canClaim =
+          assignee.isBlank() || assignee.equals(command.getValue().getAssignee());
```

Comment:
My task says to check if current assignee is `null`, but it looks to me that it will rather be an empty String.

Reply:
Yes, that's corrrect. I'll fix it. The value of the `assignee` can't be `null` since gRPC doesn't allow `null` values. I'll adjust the description.